yellow-paper/docs/public-vm/context.mdx

```diff
@@ -0,0 +1,126 @@
+| contractCallDepth     | `field`                      | Depth of the current call (how many nested calls deep is it). |
+| contractCallPointer   | `field`                      | Uniquely identifies each contract call processed by an AVM session. An initial call is assigned pointer value of 1 (expanded on in the AVM circuit section's ["Call Pointer"](./avm-circuit#call-pointer) subsection). |
+| globals               | `PublicGlobalVariables`      |  |
+| isStaticCall          | `boolean`                    |  |
+| isDelegateCall        | `boolean`                    |  |
+| calldata              | `[field; <calldata-length>]` |  |
+
+## Contract Call Results
+
+When a contract call halts, it sets the context's **contract call results** to communicate results to the caller.
+
+### _ContractCallResults_
+| Field        | Type                       | Description |
+| ---          | ---                        | ---         |
+| reverted     | `boolean`                  |  |
+| output       | `[field; <output-length>]` |  |
+
+## Context initialization
+
+### Initial contract calls
+
+An **initial contract call** initializes a new execution context from a public execution request.
+
+```
+context = AvmContext {
+    environment = INITIAL_EXECUTION_ENVIRONMENT,
+    machineState = INITIAL_MACHINE_STATE,
+    worldState = <latest world state>,
+    worldStateAccessTrace = INITIAL_WORLD_STATE_ACCESS_TRACE,
+    accruedSubstate =  { [], ... [], }, // all substate vectors empty
+    results = INITIAL_CONTRACT_CALL_RESULTS,
+}
+```
+
+> Since world state persists between transactions, the latest state is injected into a new AVM context.
+
+Given a [`PublicCallRequest`](../transactions/tx-object#public-call-request) and its parent [`TxRequest`](../transactions/local-execution#execution-request), these above-listed "`INITIAL_*`" entries are defined as follows:
+
+```
+INITIAL_EXECUTION_ENVIRONMENT = ExecutionEnvironment {
+    address: PublicCallRequest.contractAddress,
+    storageAddress: PublicCallRequest.CallContext.storageContractAddress,
+    origin: TxRequest.origin,
+    sender: PublicCallRequest.CallContext.msgSender,
+    portal: PublicCallRequest.CallContext.portalContractAddress,
+    feePerL1Gas: TxRequest.feePerL1Gas,
+    feePerL2Gas: TxRequest.feePerL2Gas,
+    feePerDaGas: TxRequest.feePerDaGas,
+    contractCallDepth: 0,
+    contractCallPointer: 1,
+    globals: <current block's global variables>
+    isStaticCall: PublicCallRequest.CallContext.isStaticCall,
+    isDelegateCall: PublicCallRequest.CallContext.isDelegateCall,
+    calldata: PublicCallRequest.args,
+    bytecode: worldState.contracts[PublicCallRequest.contractAddress],
+}
+
+INITIAL_MACHINE_STATE = MachineState {
+    l1GasLeft: TxRequest.l1GasLimit,
+    l2GasLeft: TxRequest.l2GasLimit,
+    daGasLeft: TxRequest.daGasLimit,
+    pc: 0,
+    internalCallStack: [], // initialized as empty
+    memory: [0, ..., 0],   // all 2^32 entries are initialized to zero
```

Comment:
> Memory's type is defined in the machine state definition as an array of fields....

Hmm I see... but that doesn't make much sense with the tagged memory model right? Otherwise you'd be saying that the memory is "tagged" as UINT8, but it's a field, just that it's a field that acts an int modulo 256. That's strange :) 

> What's the latest plan? To simply require that every memory cell must be initialized before it is used?

That's the current state for the simulator, but I think I can make it work for a "default 0". Just the initial typing would be awkward. The idea is that in most opcodes I have a tag, so I can do a `get(offset) := M[offset] || fromTag(tag, 0)`. This would (in the implementation) still have the mem initialized as undefined, but it would be indistinguishable from a convertible 0. I can explain in a huddle if needed but the TL;DR is that right now any uninitialized memory access yields undefined on the simulator, but I think the circuit does consider it 0.